src/app/components/cds/inputs/Select.js

```diff
@@ -26,22 +31,38 @@ const Select = ({
         { required && <span className={inputStyles.required}>*<FormattedMessage id="textfield.required" defaultMessage="Required" description="A label to indicate that a form field must be filled out" /></span>}
       </div>
     )}
-    <div className={`${inputStyles['input-container']} ${styles['select-container']}`}>
-      { iconLeft && (
-        <div className={inputStyles['input-icon-left-icon']}>
-          {iconLeft}
+    <div className={styles['input-wrapper']}>
+      <div className={`${inputStyles['input-container']} ${styles['select-container']}`}>
+        { iconLeft && (
+          <div className={inputStyles['input-icon-left-icon']}>
+            {iconLeft}
+          </div>
+        )}
+        <select
+          id="check-select__input"
+          className={`typography-body1 ${styles.input} ${variant === 'outlined' && styles.outlined} ${error && styles.error} ${iconLeft && styles['input-icon-left']}`}
+          disabled={disabled}
+          {...inputProps}
+        >
+          {children}
+        </select>
+        <div className={inputStyles['input-icon-right-icon']}>
+          <ChevronDownIcon />
         </div>
-      )}
-      <select
-        className={`typography-body1 ${styles.input} ${variant === 'outlined' && styles.outlined} ${error && styles.error} ${iconLeft && styles['input-icon-left']}`}
-        disabled={disabled}
-        {...inputProps}
-      >
-        {children}
-      </select>
-      <div className={inputStyles['input-icon-right-icon']}>
-        <ChevronDownIcon />
       </div>
+      { onRemove ?
+        <Tooltip title={<FormattedMessage id="select.removeSelection" defaultMessage="Undo selection" description="Tooltip for button on Select component to remove current selection" />}>
+          <IconButton
+            className={cx('select__clear-button', styles['select-clear-button'])}
+            onClick={() => {
+              document.getElementById('check-select__input').value = null;
```

Comment:
what do people think of this? :thinking:

Reply:
Is there any race condition that can happen between this and the `onRemove` call?

Reply:
I think it's safer to keep it out and rely on the callback.

Reply:
I don't really see a race condition, but I fixed this using a Ref. This old approach wouldn't work with multiple Selects in a page evidently.